hwtracer/src/decode/ykpt/mod.rs

```diff
@@ -19,29 +38,232 @@ impl TraceDecoder for YkPTTraceDecoder {
         let itr = YkPTBlockIterator {
             errored: false,
             parser: PacketParser::new(trace.bytes()),
+            cur_loc: ObjLoc::OtherObjOrUnknown,
+            tnts: VecDeque::new(),
+            pge: false,
         };
         Box::new(itr)
     }
 }
 
+/// Represents a location in the instruction stream of the traced binary.
+#[derive(Eq, PartialEq)]
+enum ObjLoc {
+    /// A known location in the "main binary object" of the program.
+    MainObj(u64),
+    /// Either a location not in the "main binary object", or an unknown location.
+    OtherObjOrUnknown,
+}
+
+impl Debug for ObjLoc {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::MainObj(v) => write!(f, "MainObj[0x{:x}]", v),
+            Self::OtherObjOrUnknown => write!(f, "OtherObjOrUnknown"),
+        }
+    }
+}
+
 /// Iterate over the blocks of an Intel PT trace using the fast Yk PT decoder.
 struct YkPTBlockIterator<'t> {
     /// Set to true when an error has occured.
     errored: bool,
-    /// PT packet iterator.
+    /// The packet iterator used to drive the decoding process.
     parser: PacketParser<'t>,
+    /// Keeps track of where we are in the traced binary.
+    cur_loc: ObjLoc,
+    /// A vector of "taken/not-taken" (TNT) decisions. These come in in batches and get buffered
```

Comment:
s/in in/in

Reply:
"in in" is actually correct, but I will propose a re-wording.

Reply:
[1a32fbc](https://github.com/ykjit/yk/pull/585/commits/1a32fbc45255b2d8cb31a96b9e971c480d4789e8)

Reply:
I don't think "in in" is correct, but the rephrasing is fine :)